### PR TITLE
[BugFix] Prevent TBO crash when return_logprob is enabled

### PR DESCRIPTION
**Pull request by At1a8**
## Motivation

TBO server crash after curl single prompt with `"return_logprob": true`

image

## Modifications

Clear the parent-only `extend_input_logprob_token_ids_gpu` field when splitting a `ForwardBatch` into TBO child batches.

## Accuracy Tests

100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1319/1319 [01:30<00:00, 14.58it/s]
Accuracy: 0.947
Invalid: 0.000
Latency: 90.491 s
Output throughput: 1293.545 token/s

## Speed Tests and Profiling

8k/1k conc256 np=4*conc
- w/ this PR:
  - TTT: 27472.74
  - TTFT: 33187.30ms
  - TPOT: 44.04ms
- w/o this PR:
  - TTT: 27681.44
  - TTFT: 33036.57ms
  - TPOT: 43.32ms

The code changes has almost no impact on performance.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #30144009696](https://github.com/sgl-project/sglang/actions/runs/30144009696)
Latest PR Test (Extra): :x: [Run #30144009609](https://github.com/sgl-project/sglang/actions/runs/30144009609)

### CODE REVIEW

**Review by 1am9trash — APPROVED** [review]
LGTM.
TBO's `filter_batch` guard rejects any non-None field it doesn't handle, so we need to explicitly pass `next_token_logits_buffer=None`.
Affect both NV and AMD.

**Review by kkHuang-amd — APPROVED** [review]
LGTM
